src/index.mts

```diff
@@ -0,0 +1,71 @@
+import { GatewayIntentBits } from "discord.js";
+import path from "node:path";
+import * as dotenv from "dotenv";
+import * as fs from "fs";
+import { dirname } from "path";
+import { fileURLToPath, pathToFileURL } from "url";
+import { DiscordClient } from "./discordClient.mjs";
+import { Command, Event } from "./types.mjs";
+
+dotenv.config();
+// Some hack to get __dirname to work in modules
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const client = new DiscordClient({ intents: [GatewayIntentBits.Guilds] });
+
+// Load in commands
+{
+    const foldersPath = path.join(__dirname, "commands");
+    const commandFolders = fs.readdirSync(foldersPath);
+
+    // Iterate over every command/folder
+    for (const folder of commandFolders) {
+        const commandsPath = path.join(foldersPath, folder);
+        const commandFiles = fs.readdirSync(commandsPath).filter(
+            file => file.endsWith(".mjs"), // Compiled to js!
+        );
```

Comment:
See prior statement about using a const array and allowing type inference to work it's magic.